Mathlib/CategoryTheory/GradedObject/Monoidal.lean

```diff
@@ -376,6 +528,61 @@ end Triangle
 
 end Monoidal
 
+section
+
+variable
+  [∀ (X₁ X₂ : GradedObject I C), HasTensor X₁ X₂]
+  [∀ (X₁ X₂ X₃ : GradedObject I C), HasGoodTensor₁₂Tensor X₁ X₂ X₃]
+  [∀ (X₁ X₂ X₃ : GradedObject I C), HasGoodTensorTensor₂₃ X₁ X₂ X₃]
+  [DecidableEq I] [HasInitial C]
+  [∀ X₁, PreservesColimit (Functor.empty.{0} C) ((curriedTensor C).obj X₁)]
+  [∀ X₂, PreservesColimit (Functor.empty.{0} C) ((curriedTensor C).flip.obj X₂)]
+  [∀ (X₁ X₂ X₃ X₄ : GradedObject I C), HasTensor₄ObjExt X₁ X₂ X₃ X₄]
+
+noncomputable instance monoidalCategory : MonoidalCategory (GradedObject I C) where
+  tensorObj X Y := Monoidal.tensorObj X Y
+  tensorHom f g := Monoidal.tensorHom f g
+  tensorHom_def f g := Monoidal.tensorHom_def f g
+  whiskerLeft X _ _ φ := Monoidal.whiskerLeft X φ
+  whiskerRight {_ _ φ Y} := Monoidal.whiskerRight φ Y
+  tensorUnit := Monoidal.tensorUnit
+  associator X₁ X₂ X₃ := Monoidal.associator X₁ X₂ X₃
+  associator_naturality f₁ f₂ f₃ := Monoidal.associator_naturality f₁ f₂ f₃
+  leftUnitor X := Monoidal.leftUnitor X
+  leftUnitor_naturality := Monoidal.leftUnitor_naturality
+  rightUnitor X := Monoidal.rightUnitor X
+  rightUnitor_naturality := Monoidal.rightUnitor_naturality
+  tensor_comp f₁ f₂ g₁ g₂ := Monoidal.tensor_comp f₁ g₁ f₂ g₂
+  pentagon X₁ X₂ X₃ X₄ := Monoidal.pentagon X₁ X₂ X₃ X₄
+  triangle X₁ X₂ := Monoidal.triangle X₁ X₂
+
+end
+
+section
+
+instance (n : ℕ) : Finite ((fun (i : ℕ × ℕ) => i.1 + i.2) ⁻¹' {n}) := by
+  refine Finite.of_injective (fun ⟨⟨i₁, i₂⟩, (hi : i₁ + i₂ = n)⟩ =>
+    ((⟨i₁, by omega⟩, ⟨i₂, by omega⟩) : Fin (n + 1) × Fin (n + 1) )) ?_
+  rintro ⟨⟨_, _⟩, _⟩ ⟨⟨_, _⟩, _⟩ h
+  simpa using h
+
+instance (n : ℕ) : Finite ({ i : (ℕ × ℕ × ℕ) | i.1 + i.2.1 + i.2.2 = n }) := by
+  refine Finite.of_injective (fun ⟨⟨i₁, i₂, i₃⟩, (hi : i₁ + i₂ + i₃ = n)⟩ =>
+    (⟨⟨i₁, by omega⟩, ⟨i₂, by omega⟩, ⟨i₃, by omega⟩⟩ :
+      Fin (n + 1) × Fin (n + 1) × Fin (n + 1))) ?_
+  rintro ⟨⟨_, _, _⟩, _⟩ ⟨⟨_, _, _⟩, _⟩ h
+  simpa using h
+
+-- Note: this is stated as a lemma instead of an example, because otherwise
+-- CI would complain that importing `Mathlib.CategoryTheory.Limits.Preserves.Finite`
+-- is unnecessary.
```

Comment:
So it is unnecessary, then.

I'm not going to object, but this section could become a file `GradedObject/Monoidal/Finite.lean`?